File: https:/github.com/rajesh-kayal-dev/Leetcode.git/217-ContainsDuplicate/217-ContainsDuplicate.js
// Last updated: 6/15/2026, 10:01:20 AM
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const set = new Set();

    for(const n of nums){
        if(set.has(n)) return true;
        set.add(n);
    }
    return false;
};

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 6/15/2026, 10:01:14 AM
class Solution {
    public boolean containsDuplicate(int[] nums) {

        HashSet<Integer> set = new HashSet<>();

        for(int num : nums){
            if(set.contains(num)){
                return true;
            }
            set.add(num);
        }
        return false;
    }
}